embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/api/ApiResponse.kt

```diff
@@ -47,6 +47,11 @@ sealed class ApiResponse {
      */
     data class Incomplete(val exception: Throwable) : ApiResponse()
 
+    /**
+     * API call not executed because payload was not found
+     */
+    object NoPayload : ApiResponse()
```

Comment:
No a big fan of the current abstraction. I don't want to change things for the old code, but I think the new delivery layer should differentiate between when a request has been sent and when it hasn't and have values that aren't really instances of `ApiResponse`

Reply:
Would renaming `ApiResponse` make this clearer? What other values would be needed by the SDK & how would behavior differ?

Reply:
Renaming would help. Basically the grouping is:

- Request sent
  - Response received
    - <various different classifications of responses>  
  - No response
    - Client error
    - Server error
    - Timeout
- Request not sent
  - No payload
  - Request not eligible (e.g. 429'd)
  - SDK/environment not ready (e.g. no network)
  - Client error

All of these will potentially affect when a payload is retried, and any clean up or error reporting we need to do. Renaming would work if we carefully manage all the places that check this, and deal with any fall-through states or explicitly handle new ones.

Part of me wonders if a new start is less risky.... but maybe I'll try that first and back off if it gets too hairy.